cytosnake/tests/functional/test_cli.py

```diff
@@ -0,0 +1,216 @@
+    multiple plate datasets, plate map, and no barcode. Since this is only one plate_map
+    , this means that the generated dataset came from one experiment and multiple
+    samples (plates) were used to generated the datasets.
+    """
+    # starting path
+    test_module = str(pathlib.Path().absolute())
+
+    # transfer dummy data to tmpdir
+    transfer_data(test_dir=tmp_path, n_plates=2, n_platemaps=1)
+
+    # change directory to tmpdir
+    os.chdir(tmp_path)
+
+    # execute CytoSnake
+    cmd = "cytosnake init -d *.sqlite -m metadata".split()
+    proc = subprocess.run(cmd, capture_output=True, text=True, check=False)
+
+    # leave test directory
+    os.chdir(test_module)
+
+    # clean directory,
+    cleanup_handler = CleanUpHandler(tmp_path)
+    request.addfinalizer(cleanup_handler)
+
+    # checking for success return code
+    assert proc.returncode == 0
+
+
+def test_barcode_logic_barcode_multi_platemaps(tmp_path, request) -> None:
+    """Positive case: This tests expects a successful run where the user provides
+    multiple plate datasets, multiple plate map, and barcode. Since this is only one
+    plate_map , this means that the generated dataset came from one experiment and
+    multiple samples (plates) were used to generated the datasets.
+    """
+    # PyTest module directory
+    test_module = str(pathlib.Path().absolute())
+
+    # transfer dummy data to tmpdir
+    transfer_data(test_dir=tmp_path, n_plates=2, n_platemaps=2)
+
+    # change directory to tmpdir
+    os.chdir(tmp_path)
+
+    # execute CytoSnake
+    cmd = "cytosnake init -d *.sqlite -m metadata -b barcode.txt".split()
+    proc = subprocess.run(cmd, capture_output=True, text=True, check=False)
+
+    # leave testing dir
+    os.chdir(test_module)
+
+    # clean directory,
+    cleanup_handler = CleanUpHandler(tmp_path)
+    request.addfinalizer(cleanup_handler)
+
+    # checking for success return code
+    assert proc.returncode == 0
+
+
+def test_barcode_logic_no_barcode_multi_platemaps(tmp_path, request) -> None:
+    """Negative case: This test expects a failed run where the user provides multiple
+    plate datasets, multiple plate maps (multi-experiments), and no barcode. Since
+    there are plate maps, this indicates that the generated datasets came from multiple
+    experiments.
+
+    Checks:
+    -------
+        non-zero return code
+        BarCodeRequiredError raised
+    """
+    # PyTest module directory
+    test_module = str(pathlib.Path().absolute())
+
+    # transfer dummy data to tmpdir
+    transfer_data(test_dir=tmp_path, n_plates=2, n_platemaps=2)
+
+    # change directory to tmpdir
+    os.chdir(tmp_path)
+
+    # execute CytoSnake
+    cmd = "cytosnake init -d *.sqlite -m metadata"
+    proc = subprocess.run(cmd, shell=True, capture_output=True, text=True, check=False)
+    raised_error = get_raised_error(proc.stderr)
+
+    # leave testing dir
+    os.chdir(test_module)
```

Comment:
If possible, consider sending the subdirectory information directly to CytoSnake, avoiding the need to change directory both into and out of the tmpdir. This might be my own misunderstanding, so please feel free to ignore if this isn't possible or useful in the context of this test.

Alternatively, seeing how this pattern repeats, it may be useful to create a context manager for remembering to move back to a directory after changes have occurred. See this [SO reference](https://stackoverflow.com/a/169112) for one example.

Reply:
Thanks for the resource. This will be considered in #49